### PR TITLE
fix: use inclusive boundary for gossip clock disparity checks

### PR DESCRIPTION
**Pull request by lodekeeper**
## Description

Fix off-by-one in gossip clock disparity boundary checks. The clock used strict less-than (`<`) instead of less-than-or-equal (`<=`) when comparing time differences against `MAXIMUM_GOSSIP_CLOCK_DISPARITY`, causing messages arriving at exactly the disparity boundary to be incorrectly rejected.

### Spec Reference

The [consensus spec](https://github.com/ethereum/consensus-specs/blob/dev/specs/phase0/p2p-interface.md#beacon_block) defines the future slot check as:

```python
block_time_ms = compute_time_at_slot_ms(state, block.slot)
if current_time_ms + MAXIMUM_GOSSIP_CLOCK_DISPARITY < block_time_ms:
    raise GossipIgnore("block is from a future slot")
```

The spec uses strict `<` for the **rejection** condition, meaning a message should be **accepted** when:

```
current_time_ms + MAXIMUM_GOSSIP_CLOCK_DISPARITY >= block_time_ms
```

Which is equivalent to:

```
block_time_ms - current_time_ms <= MAXIMUM_GOSSIP_CLOCK_DISPARITY
```

### The Bug

Our clock was checking `nextSlotTime - Date.now() < MAXIMUM_GOSSIP_CLOCK_DISPARITY` (strict `<`), but the spec requires `<=` (inclusive). At exactly the boundary (e.g. exactly 500ms before the next slot), the clock would reject the message as a future slot when the spec says it should be accepted.

This was caught by the [gossip validation spec tests](https://github.com/ethereum/consensus-specs/tree/dev/tests/minimal/phase0/networking/gossip_beacon_block) which include a test pair that validates the exact boundary:
- `gossip_beacon_block__valid_within_clock_disparity`: `current_time_ms: 5500`, block at slot 1 (6000ms) → gap = 500ms → **expected: valid**
- `gossip_beacon_block__ignore_future_slot`: `current_time_ms: 5499`, same block → gap = 501ms → **expected: ignore**

### Changes

Changed `<` to `<=` in three places in `Clock`:
1. `currentSlotWithGossipDisparity` getter
2. `isCurrentSlotGivenGossipDisparity` — next slot boundary check
3. `isCurrentSlotGivenGossipDisparity` — previous slot boundary check

### Practical Impact

Minimal — this only affects messages arriving at the exact millisecond boundary of the disparity window. But it aligns our implementation with the spec.

---

*This PR was authored with AI assistance (Lodekeeper 🌟).*

### CODE REVIEW

File: packages/beacon-node/src/util/clock.ts [hunk 1]
```diff
@@ -129,12 +129,12 @@ export class Clock extends EventEmitter implements IClock {
...
+    if (nextSlotTime - Date.now() <= this.config.MAXIMUM_GOSSIP_CLOCK_DISPARITY) {
       return slot === currentSlot + 1;
     }
     const currentSlotTime = computeTimeAtSlot(this.config, currentSlot, this.genesisTime) * 1000;
     // we've just passed the current slot, accept previous slot
-    if (Date.now() - currentSlotTime < this.config.MAXIMUM_GOSSIP_CLOCK_DISPARITY) {
+    if (Date.now() - currentSlotTime <= this.config.MAXIMUM_GOSSIP_CLOCK_DISPARITY) {
       return slot === currentSlot - 1;
     }
     return false;
```
[the remark this answers is not in the document]
Comment (lodekeeper) [reply packages/beacon-node/src/util/clock.ts:145]: @gemini-code-assist Good catch — the multiple `Date.now()` calls in `isCurrentSlotGivenGossipDisparity` could theoretically produce inconsistent results if the millisecond ticks between calls. That said, this is pre-existing behavior and not introduced by this PR (this PR only changes `<` to `<=`). I'd prefer to keep this PR minimal and focused on the boundary fix. Happy to address the `Date.now()` consolidation in a follow-up if reviewers want it.

**Review by nflaig — DISMISSED** [review]

File: packages/beacon-node/src/util/clock.ts [hunk 2]
```diff
@@ -100,7 +100,7 @@ export class Clock extends EventEmitter implements IClock {
   get currentSlotWithGossipDisparity(): Slot {
     const currentSlot = this.currentSlot;
     const nextSlotTime = computeTimeAtSlot(this.config, currentSlot + 1, this.genesisTime) * 1000;
-    return nextSlotTime - Date.now() < this.config.MAXIMUM_GOSSIP_CLOCK_DISPARITY ? currentSlot + 1 : currentSlot;
+    return nextSlotTime - Date.now() <= this.config.MAXIMUM_GOSSIP_CLOCK_DISPARITY ? currentSlot + 1 : currentSlot;
```
Comment (twoeths) [inline packages/beacon-node/src/util/clock.ts:105]: @lodekeeper  logic looks correct but would like to add a comment for it referring to the spec here and at below changes

Comment (lodekeeper) [reply packages/beacon-node/src/util/clock.ts:105]: Added spec reference comments to both `currentSlotWithGossipDisparity` and `isCurrentSlotGivenGossipDisparity`. The spec (phase0/p2p-interface.md) uses strict `<` for rejection (`current_time + MAXIMUM_GOSSIP_CLOCK_DISPARITY < message_time`), so the boundary case (exactly equal) is accepted — hence `<=` here.

**Review by twoeths — APPROVED** [review]

**Review by nflaig — APPROVED** [review]

**Comment by wemeetagain** [comment]
:tada: This PR is included in [v1.41.0](https://github.com/ChainSafe/lodestar/releases/tag/v1.41.0) :tada:
